consensus/spos/bls/v2/subroundSignature.go

```diff
@@ -488,16 +301,13 @@ func (sr *subroundSignature) doSignatureJobForSingleKey(isSelfLeader bool, isFla
 		return false
 	}
 
-	// leader already sent his signature on subround block
-	if !isSelfLeader {
-		ok := sr.createAndSendSignatureMessage(signatureShare, []byte(sr.SelfPubKey()))
-		if !ok {
-			return false
-		}
+	// leader also sends his signature here
```

Comment:
shouldn't we skip it for the leader and use the one sent with the block?

Reply:
if we reuse the signature from the original proposed header (which now is on the LeaderSignature field, as there is no longer a consensus message to hold also in a different field the leader signature) then we would need to update the verification and signing methods to remove the signature from the block before signing or verifying the proof

Reply:
added a todo also to check again the block proof verification & leader signature verification